### PR TITLE
Refactor recording & reporting agent time usage

### PR DESCRIPTION
**Pull request by satococoa**
## Summary
Refactor recording & reporting agent time usage.

## Related Issue
https://github.com/giselles-ai/giselle/pull/251/files#r1891016983

> I would like to separate the recording of the execution from the context of the execution, so it would be better to make it a callback like onFinishPerformFlowExecution and add props like onFinishPerformFlowExecutionAction in the Provider. I think, but in this Pull request, it is fine as it is.

### CODE REVIEW

File: app/(playground)/p/[agentId]/contexts/execution.tsx [hunk 1]
```diff
@@ -552,16 +557,10 @@ export function ExecutionProvider({
...
 				},
 			});
-
-			await recordAgentUsageAction(
-				currentExecution.runStartedAt,
-				runEndedAt,
-				currentExecution.durationMs,
-			);
-
+			await onFinishPerformExecution(runEndedAt, totalFlowDurationMs);
```
Comment (toyamarinyon) [inline app/(playground)/p/[agentId]/contexts/execution.tsx]:
```suggestion
			await onFinishPerformExecution?.(runEndedAt, totalFlowDurationMs);
```

File: app/(playground)/p/[agentId]/contexts/execution.tsx [hunk 2]
```diff
@@ -611,10 +610,27 @@ export function ExecutionProvider({
...
+				onFinishPerformExecution: async (
+					endedAt: number,
+					totalDurationMs: number,
+				) => {
+					await onFinishPerformExecutionAction(
+						flowRunStartedAt,
+						endedAt,
+						totalDurationMs,
+					);
+				},
```
Comment (toyamarinyon) [inline app/(playground)/p/[agentId]/contexts/execution.tsx]:
```suggestion
				onFinishPerformExecution: (endedAt, durationMs) =>
					onFinishPerformExecutionAction(flowRunStartedAt, endedAt, durationMs),

```

File: app/(playground)/p/[agentId]/contexts/execution.tsx [hunk 3]
```diff
@@ -670,10 +686,23 @@ export function ExecutionProvider({
 						};
 					});
 				},
+				onFinishPerformExecution: async (endedAt, durationMs) => {
+					await onFinishPerformExecutionAction(
+						flowRunStartedAt,
+						endedAt,
+						durationMs,
+					);
+				},
```
Comment (toyamarinyon) [inline app/(playground)/p/[agentId]/contexts/execution.tsx]:
```suggestion
				onFinishPerformExecution: (endedAt, durationMs) =>
					onFinishPerformExecutionAction(flowRunStartedAt, endedAt, durationMs),

```

File: app/(playground)/p/[agentId]/contexts/execution.tsx [hunk 4]
```diff
@@ -734,6 +763,13 @@ export function ExecutionProvider({
 						},
 					});
 				},
+				onFinishPerformExecution: async (endedAt, durationMs) => {
+					await onFinishPerformExecutionAction(
+						flowRunStartedAt,
+						endedAt,
+						durationMs,
+					);
+				},
```
Comment (toyamarinyon) [inline app/(playground)/p/[agentId]/contexts/execution.tsx]:
```suggestion
				onFinishPerformExecution: (endedAt, durationMs) =>
					onFinishPerformExecutionAction(flowRunStartedAt, endedAt, durationMs),

```

File: app/(playground)/p/[agentId]/contexts/execution.tsx [hunk 5]
```diff
@@ -448,6 +448,10 @@ export function ExecutionProvider({
 			stepExecution: CompletedStepExecution,
 			artifact: TextArtifact,
 		) => void;
+		onFinishPerformExecution: (
+			endedAt: number,
+			durationMs: number,
+		) => Promise<void>;
```
Comment (toyamarinyon) [inline app/(playground)/p/[agentId]/contexts/execution.tsx]: I think this should be Optional.

```suggestion
		onFinishPerformExecution?: (
			endedAt: number,
			durationMs: number,
		) => Promise<void>;
```

**Review by toyamarinyon — APPROVED** [review]
Thank you! We are proposing some minor changes, please check them and apply them if necessary.

**Review by shige — APPROVED** [review]
LGTM! ♻️

**Pull request merged**